src/contexts/Providers.jsx

```diff
@@ -2,6 +2,7 @@ import { SellProvider } from './SellContext';
 import { AuthProvider } from './AuthContext';
 
 // TODO move the sell provider to just the /sell routes
+// TODO move the buy provide to just the /buy routes
```

Comment:
im thinking about this more, and I actually might not care if the provider is accessible across all components. Only the relevant components would probably access the providers they need so this extra work might not be worth

Reply:
It doesn't make sense to me but when I was trying to debug the usecontext I moved the BuyContext out of the Providers file and I don't know if that was part of the solution to fix my context issue. Either way I think it would be good to keep the BuyContext only relevant to the routes that need it for abstraction

Reply:
ok that is fair. In that case, Im gunna push that we do one or the other here and either move the provider just around the buy routes or just wrap it all tgt using the generic providers to keep things a bit cleaner